fireplace/cards/utils.py

```diff
@@ -39,52 +41,40 @@
 
 
 # take a list of weights and a list of card pools and produce a random weighted sample without replacement
-def weighted_card_choice(source, weights, card_sets, count):
+# len(weights) == len(card_sets) (one weight per card set)
+def weighted_card_choice(source, weights: List[int], card_sets: List[str], count: int):
```

Comment:
Make weights default to `None` and check it in the short-circuit; also make it the last arg